src/Package/TestApp/Help/HelpWindowTest.cs

```diff
@@ -68,36 +62,41 @@ public class HelpWindowTest : HostBasedInteractiveTest {
                 clientApp.Uri.PathAndQuery.Should().Be("/library/stats/html/lm.html");
 
                 await ExecCommandAsync(clientApp, RPackageCommandId.icmdHelpHome);
+                await WaitForReadyAndRenderedAsync(clientApp);
                 clientApp.Uri.PathAndQuery.Should().Be("/doc/html/index.html");
             }
         }
 
         private async Task ShowHelpAsync(string command, VsRHostScript hostScript, RHostClientHelpTestApp clientApp) {
+            clientApp.ResetReadyState();
             await hostScript.Session.ExecuteAsync($"rtvs:::show_help({command.ToRStringLiteral()})").SilenceException<RException>();
-            await clientApp.Ready.Task;
+            await WaitForReadyAndRenderedAsync(clientApp);
         }
 
         private async Task ExecCommandAsync(RHostClientHelpTestApp clientApp, int commandId) {
             UIThreadHelper.Instance.Invoke(() => {
                 object o = new object();
                 clientApp.Component.Controller.Invoke(RGuidList.RCmdSetGuid, commandId, null, ref o);
             });
-            await clientApp.Ready.Task;
+            await clientApp.Ready;
         }
 
-        private async Task<string> GetBackgroundColorAsync(IHelpVisualComponent component) {
+        private async Task<string> GetBackgroundColorAsync(IHelpVisualComponent component, RHostClientHelpTestApp clientApp) {
             string color = "red";
 
-            await UIThreadHelper.Instance.InvokeAsync(async () => {
-                while (component.Browser.ReadyState != WebBrowserReadyState.Complete) {
-                    await UIThreadHelper.Instance.DoEventsAsync();
-                }
-                await component.ReadyAsync;
-
+            await WaitForReadyAndRenderedAsync(clientApp);
+            await UIThreadHelper.Instance.InvokeAsync(() => {
                 IHTMLElement2 body = component.Browser.Document.Body.DomElement as IHTMLElement2;
                 color = body.currentStyle.backgroundColor as string;
             });
+
             return color;
         }
+
+        private async Task WaitForReadyAndRenderedAsync(RHostClientHelpTestApp clientApp) {
+            await UIThreadHelper.Instance.InvokeAsync(() => DoIdle(500));
```

Comment:
Browser needs UI thread time to complete rendering. It is old IE it does things via PostMessage